packages/components/src/Dropdown/Dropdown.tsx

```diff
@@ -48,6 +55,23 @@ type DropdownProps = ListboxProps & {
    * Optional className for additional styling.
    */
   className?: string;
+  /**
+   * Render dropdown button that is only as wide as the content.
+   *
+   * When defining compact dropdown need to provide optionsWidth to
+   * define width of options menu, and optionally provide optionsAlign
+   * if desired to right align to dropdow button.
+   */
+  compact?: boolean;
+  /**
+   * Align dropdown menu to the left (default) or right of dropdown button
+   */
+  optionsAlign?: OptionsAlignType;
+  /**
+   * Render dropdown menu with width (specify using tailwind utility string)
+   * independent of dropdown button width
+   */
+  optionsWidth?: string;
```

Comment:
Do these need to be separate props independent of `compact`? i.e.:
- Will a `compact` dropdown always have right aligned options and a specific width?
- Or, is this accounting for possible flexibility with the Dropdown options, independent of what the button looks like?

If the first, I'd recommend removing the external props and enforcing the styling for `compact`. If the latter, that's basically what the existing Dropdown is already -- our advice (to avoid complicating the existing component), is to apply your own Tailwind classes on `Dropdown.Options` if you need to customize the styles further.

Does that make sense? Or we could surface `optionsClassname` in the overall Dropdown if you anticipate _every_ `compact` Dropdown is going to need some customization

Reply:
Thanks for the review @anniehu4 .

So, users of compact variant will always need to style the options menu.

The first reason is that compact sizes to the length of the selected option text and then the options menu sizes according to that button width. So if options are: { 'abc', 'abcde', 'abcdefg' } where 'abc' is selected, the options menu would open up too narrow to display text. I suppose we might calculate the widest option and pass that back to size the button, but that seems overly complicated.

As for left and right aligned, it is the same problem as described above. If 'abc' is selected then the options menu will need to be wider and will stick out. Depending on context (ex: if the dropdown button is on the left of the comment card or to the right), we'll want the options menu to stick out to the left or to the right.

Given that, I like to have specific props that tells the developer they need to specify at least width if they selected compact. Also on a side note I was having trouble with `w-full` being applied to options and that seemed to take precedence over classname provided to Dropdown.Options.

That said, imo we can be specific with this variant, or maybe not add this variant as developer can already (I think) do this with composition and providing classname in Dropdown.Options. I'm on the fence on this and would appreciate your thoughts on this. Thanks!

Reply:
hi @silvestry! sorry, catching up from HQ week still. thanks for explaining the behavior of this variant more. Some of this styling does seem a bit specific (especially for the right aligned options), so I think it's worth handling it here and saving some dev effort. Based on what you said, I'd consider:
- keeping the `optionsAlign` prop 👍 
- renaming `optionsWidth` to `optionsClassname`
    - the reasoning here is EDS should work independently of a consumer app using tailwind or not, so it's a little strange to specify that a prop must pass in a tailwind utility class. `optionsClassname` still would handle a utility class, but can also handle a regular className -- which may be helpful extra flexibility anyways. We can document the custom width use case in storybook + zeroheight, and I'm assuming devs are testing the changes they're making and would notice if custom width is needed. If you're really worried, we could throw a console warning if this prop isn't provided when `"compact"` is
    - a different option could be keeping it named `optionsWidth` but either:
        - taking a raw size value (like `"6rem"`), and passing that through to CSS
        - setting a union of possible values (may be annoying to maintain)

Reply:
@anniehu4 : Cool. Sounds good. I've kept `optionsAlign` prop and renamed `optionsWidth` to `optionsClassName` (notice the casing... not sure what is the standard in EDS). I've also added check to ensure `optionsClassName` under the `compact` variant.

Reply:
Actually, given @dierat 's comment below, I'll remove the check as `optionsClassName` is not necessary with `compact`.